runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -411,49 +400,57 @@ public void stop(boolean shutdown) {
      *     or a timeout in the case there is no response.
      */
     public <T> CompletableFuture<T> sendMessageAndGetCompletable(ChannelHandlerContext ctx,
-        CorfuMsg message) {
+        @NonNull CorfuMsg message) {
         boolean isEnabled = MetricsUtils.isMetricsCollectionEnabled();
-        // Check the connection future, and wait for it to be successful before
-        // sending the message.
+
+        // Check the connection future. If connected, continue with sending the message.
+        // If timed out, return a exceptionally completed with the timeout.
         try {
             connectionFuture
                 .get(parameters.getConnectionTimeout().toMillis(), TimeUnit.MILLISECONDS);
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             throw new UnrecoverableCorfuInterruptedError(e);
         } catch (TimeoutException | ExecutionException e) {
-            // If we timed out, return a CompletableFuture exceptionally completed
-            // with the timeout.
             CompletableFuture<T> f = new CompletableFuture<>();
             f.completeExceptionally(e);
             return f;
         }
-        final Timer.Context context = MetricsUtils
-            .getConditionalContext(isEnabled, timerSyncOp);
+
+        // Set up the timer and context to measure request
+        final Timer roundTripMsgTimer = getTimer(message);
+        final Timer.Context roundTripMsgContext = MetricsUtils
+                .getConditionalContext(isEnabled, roundTripMsgTimer);
+
         // Get the next request ID.
         final long thisRequest = requestID.getAndIncrement();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "requestID"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)